packages/shorebird_cli/lib/src/commands/preview_command.dart

```diff
@@ -308,4 +341,84 @@ class PreviewCommand extends ShorebirdCommand {
       '${platform.name}_${release.version}.$extension',
     );
   }
+
+  Future<void> setChannelOnRunner({
+    required Directory runnerDirectory,
+    required String channel,
+  }) async {
+    await Isolate.run(() async {
+      final shorebirdYaml = File(
+        p.join(
+          runnerDirectory.path,
+          'Frameworks',
+          'App.framework',
+          'flutter_assets',
+          'shorebird.yaml',
+        ),
+      );
+
+      if (!shorebirdYaml.existsSync()) {
+        throw Exception('Unable to find shorebird.yaml');
+      }
+
+      final yaml = YamlEditor(shorebirdYaml.readAsStringSync())
+        ..update(['channel'], channel);
+
+      shorebirdYaml.writeAsStringSync(yaml.toString(), flush: true);
+    });
+  }
+
+  Future<void> setChannelOnAab({
+    required File aabFile,
+    required String channel,
+  }) async {
+    // Getting the reference here since we cannot inside the isolate.
+    final extractZip = artifactManager.extractZip;
+
+    await Isolate.run(() async {
+      final tempDir = Directory.systemTemp.createTempSync();
+      final basename = p.basenameWithoutExtension(aabFile.path);
+      final outputPath = p.join(tempDir.path, basename);
+
+      await extractZip(
+        zipFile: aabFile,
+        outputDirectory: Directory(outputPath),
+      );
+
+      final shorebirdYaml = File(
+        p.join(
+          outputPath,
+          'base',
+          'assets',
+          'flutter_assets',
+          'shorebird.yaml',
+        ),
+      );
+
+      if (!shorebirdYaml.existsSync()) {
+        throw Exception('Unable to find shorebird.yaml');
+      }
+
+      final yaml = YamlEditor(shorebirdYaml.readAsStringSync())
+        ..update(['channel'], channel);
+
+      shorebirdYaml.writeAsStringSync(yaml.toString(), flush: true);
+
+      // This is equivalent to `zip --no-dir-entries`
+      // Which does NOT create entries in the zip archive for directories.
+      // It's important to do this because bundletool expects the
+      // .aab not to contain any directories.
+      final encoder = ZipFileEncoder()..create(aabFile.path);
+      for (final file in Directory(outputPath).listSync(recursive: true)) {
+        if (file is File) {
+          await encoder.addFile(
+            file,
+            file.path.replaceFirst('$outputPath/', ''),
+          );
+        }
+      }
```

Comment:
Not necessary, but a (maybe) more readable way of writing this:

```dart
Directory(outputPath).listSync(recursive: true))
  .whereType<File>()
  .forEach((file) async => 
    await encoder.addFile(
      file,
      file.path.replaceFirst('$outputPath/', ''),
    ),
  );
```

Reply:
I just didn't like the having to iterate through the list twice